src/s2wasm.h

```diff
@@ -834,6 +845,34 @@ class S2WasmBuilder {
       curr->finalize();
       setOutput(curr, assign);
     };
+    auto fixRelocation = [&](LinkerObject::Relocation* relocation,
```

Comment:
Somewhere we should actually describe how imported globals work. Also it's a little weird that this function called `fixRelocation` doesn't really do anything to the relocation. Instead it either adds it unmodified, or generates some code (at which point it will never get added?). `addRelocation` might be marginally better; it seems like we could do even better than that, but it's not immediately obvious how).
Since we already handle null relocations, we could separate the `addRelocation` bit and put the interesting logic in some function that either returns the relocation unchanged, or generates code and returns nullptr? Not sure.